### ISSUE TITLE
Agent for scheduled list of static events

### ISSUE THREAD

**Issue by Shalmezad** [cantino/huginn#1936]
Let's say we have an web agent that can fetch comments from a page like `:username/comments`, and we want to run it every 12 hours for a fixed list of usernames, so would need some sort of generator feeding a list of username via events.

A manual event works, but is currently unable to be scheduled.
A CSV list works, but requires setting up a CSV file with all the data you want to send, and getting the formatting just right.
A javascript agent works (don't think there's anything that can't be solved with it), but requires knowing javascript to get working.

Per @cantino's suggestion:
```
Maybe the ManualEventAgent should be extended to event any Static event and could be scheduled?
```

**Comment by dsander** [comment]
I like the idea and think a similar request came up in the past, but I can't find the issue at the moment.

**Comment by knu** [comment]
That was discussed around here: https://github.com/cantino/huginn/issues/353#issuecomment-269685050

**Comment by cantino** [comment]
I do think a simple Agent that will emit a custom event on a schedule would be a useful addition.

**Comment by dsander** [comment]
@knu Thanks

@cantino Do you think this should be a new Agent?

**Comment by cantino** [comment]
I do, yes. If someone wants to add it, go for it. Otherwise I'll try to get to it in the next few weeks.

**Comment by cantino** [comment]
Sorry, I haven't been able to get to this. I've added the `help-wanted` label.

**Comment by nogre** [comment]
I have a potential (hack) solution for this: Create an Event Formatting Agent with all the fields that you want in the `instructions`. Then set a Manual agent to run it on command. To schedule it, make a JavaScript Agent that feeds into it with, say,
Agent.check = function() {this.createEvent({ 'message': 'I made an event!' });};
as its content. 

Else you could just allow Event Formatting Agents to be scheduled. (also left this comment [here](https://github.com/huginn/huginn/issues/1943#issuecomment-301332821))

**Comment by JohnMaguire** [comment]
Surprised to see that SchedulerAgent cannot create events and Manual Event Agent cannot be scheduled. This makes it hard to do things like schedule a reminder every week or month, etc.

**Comment by virtadpt** [comment]
I agree with you, it is somewhat counterintuitive.  I usually work around it by having SchedulerAgent trigger a MessageFormattingAgent that emits a static event.

Maybe a "SendAnEventAgent" that just emits a static string would be a solution?

**Comment by JohnMaguire** [comment]
@virtadpt As someone new to Huginn, this has actually been a point of confusion for me - there are lots of "Agents" but they seem to do different things... at least three that I can determine:

1. *Trigger* another agent
2. Emit an event
3. Receive an event

All three of these are referred to as "agents." It might be nice to know what's a "producer" or "consumer." But the hardest part to wrap my mind around is _triggering_. Can't events be triggered by receiving an event? How is this different than being triggered via a schedule, for example? And finally, what's the difference between the "Schedule" option and SchedulerAgent?

I'd be happy to try to improve docs around this area if I had a better understanding of Huginn conceptually.

**Comment by virtadpt** [comment]
You are correct - all of the Agents carry out different tasks.  They're optimized for specific things, kind of like Tinkertoys.

There is a fourth thing that Agents do, which is contact an external service and emit whatever it gets as an event (if it's not an error).

I don't think there's a comprehensive list of Agents that can be triggered by the Scheduler Agent.  I can only list the ones that I use in that fashion, which is far from comprehensive.

The icon to differentiate them on the overview page says which Scenario a given Agent is associated with (if any).  It would be nice if they had individual icons but I don't think that's anywhere on the roadmap right now.

It's easier to list the ones that _aren't_ producers or consumers because they're the exceptions and not the rules.
